### ISSUE TITLE
harp_program_from_string_fuzzer: Integer-overflow in harp_parse_double

### ISSUE THREAD

**Issue by schwehr** [stcorp/harp#216]
```
third_party/stcorp_harp/libharp/harp-utils.c:959:33: runtime error: signed integer overflow: 10 * 3333333333333333333 cannot be represented in type 'long'
    #0 0x5590fce7594c in harp_parse_double third_party/stcorp_harp/libharp/harp-utils.c:959:33
    #1 0x5590fce28811 in harp_operation_parser_parse libharp/harp-operation-parser.y:323:17
    #2 0x5590fce33d05 in harp_program_from_string libharp/harp-operation-parser.y:1656:9
    #3 0x5590fcc35b8c in LLVMFuzzerTestOneInput third_party/stcorp_harp/fuzz/harp_program_from_string_fuzzer.cc:19:3
    #4 0x5590fda9c486 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) third_party/llvm/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:556:15
 ```

```c
        while (length > 0)
        {
            if (*buffer < '0' || *buffer > '9')
            {
                break;
            }
            exponent_value = 10 * exponent_value + (*buffer - '0');  // <--- here
            exponent_length++;
            buffer++;
            length--;
        }
```

Using 

```c++
#include <stddef.h>
#include <stdint.h>

#include <string>

#include "third_party/absl/cleanup/cleanup.h"
#include "third_party/stcorp_harp/libharp/harp.h"
extern "C" {
#include "third_party/stcorp_harp/libharp/harp-program.h"
}  // extern "C"

extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
  const std::string operations(reinterpret_cast<const char *>(data), size);

  harp_init();
  auto done = absl::MakeCleanup([] { harp_done(); });

  harp_program *program;
  harp_program_from_string(operations.c_str(), &program);

  return 0;
}
```

[testcase-5983867848097792.zip](https://github.com/stcorp/harp/files/4520494/testcase-5983867848097792.zip)

**Comment by svniemeijer** [comment]
Should be fixed in 06df45ec6eef8bd235181f699dd7b1478cfa7210
(similar fix has also been applied in CODA)

**Comment by schwehr** [comment]
Verified
